implement-directus-auth-in-nextjs-14/index.md

```diff
@@ -0,0 +1,995 @@
+          <>
+            <form action={handleLogout}>
+              <button>Sign out</button>
+            </form>
+          </>
+        ) : (
+          <NavLink item={{ title: "Sign in", path: "/login" }} />
+        )}
+      </div>
+    </div>
+  );
+};
+export default Links;
+```
+
+In `links` create a `navLink` directory with `navLink.jsx` and add the following:
+
+```javascript
+"use client";
+import Link from "next/link";
+
+const NavLink = ({ item }) => {
+
+  return (
+    <Link
+      href={item.path}
+    >
+      {item.title}
+    </Link>
+  );
+};
+export default NavLink;
+```
+
+To render the `Navbar` component on all pages, update `Layout.js` as follows:
+
+```javascript
+import Navbar from "@/components/navbar/Navbar";
+
+export const metadata = {
+  title: {
+    default:"Next.js 14 Homepage",
+  },
+  description: "Next.js Directus app",
+};
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <body>
+          <div>
+            <Navbar />
+            {children}
+          </div>
+      </body>
+    </html>
+  );
+}
+```
+
+### Loader Component
+
+In the `app` directory, create `Loading.jsx` with the content:
+
+```javascript
+const Loading = () => {
+  return (
+    <div>Loading</div>
+  )
+}
+export default Loading
+```
+
+### Not Found Component
+
+In the `app` directory, create `not-found.jsx` with the content:
+
+```javascript
+//src/app/not-found.jsx
+import Link from "next/link"
+const NotFound = () => {
+  return (
+    <div>
+      <h2>Not Found</h2>
+      <p>Sorry, the page you are looking for does not exist.</p>
+      <Link href="/">Return Home</Link>
+    </div>
+  )
+}
+export default NotFound
+```
+
+## Route Protections
+
+To handle route protection, Create the `post/layout.jsx` file and add the following:
+
+```javascript
+"use client"
+import { userSession } from "@/lib/action";
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+
+export default function PostLayout({ children }) {
+  const [isSuccess, setIsSuccess] = useState(false);
+  const router = useRouter();
+
+  useEffect(() => {
+    (async () => {
+      const user = await userSession();
+
+      if (!user.accessToken) {
+        router.push("/login");
+        return;
+      }
+
+      setIsSuccess(true);
+    })();
+  }, [router]);
+
+  if (!isSuccess) {
+    return <p>Loading...</p>;
+  }
+  return <div>{children}</div>;
+}
+```
+
+This will ensure that only authenticated users can access the create post and update post pages.
+
+## Implementing Server Actions for Auth Components
+
+Server Actions are **asynchronous functions** that are executed on the server. They can be used in Server and Client Components to handle form submissions and data mutations in Next.js applications.
+
+To send user credentials to the `Directus` backend on the user register form submission, navigate to the `lib` directory, create `action.js` file and add the following:
```

Comment:
Can you please start this as a minimal file. And then add functions as we get to their respective sections?